### ISSUE TITLE
Heroku fails to install zappa dependency via npm

### ISSUE THREAD

**Issue by dimroc** [mauricemach/zappa#57]
A git push heroku command ultimate results in the following snippet (perhaps it's an npm bug):

```
   > 
```

-----> Heroku receiving push
-----> Node.js app detected
-----> Vendoring node 0.4.7
-----> Installing dependencies with npm 1.0.8

```
   > mongodb@0.9.6-7 install /tmp/build_34k61587jt8zz/node_modules/mongoose/node_modules/mongodb
   > bash ./install.sh

   ================================================================================
   =                                                                              =
   =  To install with C++ bson parser do <npm install mongodb --mongodb:native>   =
   =  the parser only works for node 0.4.X or lower                               =
   =                                                                              =
   ================================================================================
   Not building native library for cygwin
   npm ERR! error installing zappa@0.2.0beta Error: ENOENT, No such file or directory '/tmp/build_34k61587jt8zz/node_modules/zappa/bin/zappa'
   npm ERR! Error: ENOENT, No such file or directory '/tmp/build_34k61587jt8zz/node_modules/zappa/bin/zappa'
   npm ERR! Report this *entire* log at:
   npm ERR!     <http://github.com/isaacs/npm/issues>
   npm ERR! or email it to:
   npm ERR!     <npm-@googlegroups.com>
   npm ERR! 
   npm ERR! System Linux 2.6.18-xenU-ec2-v1.2
   npm ERR! command "/mnt/slug-compiler/language_packs/node/vendor/node/node-0.4.7/node" "/mnt/slug-compiler/language_packs/node/vendor/npm/npm-1.0.8/cli.js" "install"
   npm ERR! 
   npm ERR! Additional logging details can be found in:
   npm ERR!     /tmp/build_34k61587jt8zz/npm-debug.log
   npm not ok
```

 !     Failed to install dependencies with npm
 !     Heroku push rejected, failed to compile Node.js app

**Comment by smathy** [comment]
Looks like one of the other libs is expecting the 0.1.x version of zappa (the one with the `bin/zappa` command), whereas npm is getting/using the 0.2.0beta version which no longer has that executable.

**Comment by mauricemach** [comment]
I setup a test app to investigate your issue and #59, but could only reproduce the latter. After a quick fix for that issue I was able to deploy the app.

@smathy is right, it's expecting the zappa bin whereas 0.2.0beta doesn't have it anymore, which doesn't make sense to me.

I followed [these instructions](http://devcenter.heroku.com/articles/node-js) to deploy it. Do you find any relevant difference in your setup vs the example? Does it work locally with foreman?

**Comment by dimroc** [comment]
I also was unable to repro it when deploying to a fresh heroku app git destination. The issue seems to surface when upgrading an existing Heroku app from zappa 0.1.5 -> 0.2.0beta. I was able to do this locally with npm 1.0.13 but when pushing to heroku, which uses npm 1.0.8, I get the issue described above.

I'll probably go ahead and just destroy my heroku app and recreate it (is there an easy way to flush out npm dependencies on Heroku? Heroku run npm doesn't work.) to move on, but still think this is something to be aware of. 

Rather not have to recreate the heroku app when upgrading. I have opened a ticket with heroku asking them to upgrade npm because this could be an npm issue.

Thanks for the feedback!
